lib/src/mock_snapshot_metadata.dart

```diff
@@ -0,0 +1,21 @@
+import 'package:cloud_firestore/cloud_firestore.dart';
+import 'package:mockito/mockito.dart';
+
+class MockSnapshotMetadata extends Mock implements SnapshotMetadata {
+  bool _hasPendingWrites;
+  bool _isFromCache;
+
+  MockSnapshotMetadata({
+    hasPendingWrites = false,
+    isFromCache = false,
+  }) {
+    _hasPendingWrites = hasPendingWrites;
+    _isFromCache = isFromCache;
+  }
+
+  @override
+  bool get hasPendingWrites => _hasPendingWrites;
```

Comment:
Same, I'd prefer to simply write it as :

```dart
class MockSnapshotMetadata extends Mock implements SnapshotMetadata {
  @override
  bool get hasPendingWrites => false;

  @override
  bool get isFromCache => false;
}
```

Until someone implements custom metadata values.

Reply:
I changed this to simply the method. In the future, maybe I'll send another PR to enable set metadata mocks. Thanks